modules/prebidServerBidAdapter/index.js

```diff
@@ -540,35 +552,37 @@ const OPEN_RTB_PROTOCOL = {
       if (!request.user.ext.tpid) {
         request.user.ext.tpid = {}
       }
-      Object.assign(request.user.ext.tpid, bidRequests[0].userId);
-    }
 
-    if (bidRequests && bidRequests[0].gdprConsent) {
-      // note - gdprApplies & consentString may be undefined in certain use-cases for consentManagement module
-      let gdprApplies;
-      if (typeof bidRequests[0].gdprConsent.gdprApplies === 'boolean') {
-        gdprApplies = bidRequests[0].gdprConsent.gdprApplies ? 1 : 0;
-      }
-
-      if (request.regs) {
-        if (request.regs.ext) {
-          request.regs.ext.gdpr = gdprApplies;
+      const gdprConsent = utils.deepAccess(bidRequests, '0.gdprConsent');
+      if (gdprConsent) {
+        // note - gdprApplies & consentString may be undefined in certain use-cases for consentManagement module
+        const gdprApplies = (typeof gdprConsent.gdprApplies === 'boolean') ? Number(gdprConsent.gdprApplies) : undefined;
+        if (request.regs) {
+          if (request.regs.ext) {
+            request.regs.ext.gdpr = gdprApplies;
+          } else {
+            request.regs.ext = { gdpr: gdprApplies };
+          }
         } else {
-          request.regs.ext = { gdpr: gdprApplies };
+          request.regs = { ext: { gdpr: gdprApplies } };
         }
-      } else {
-        request.regs = { ext: { gdpr: gdprApplies } };
-      }
 
-      let consentString = bidRequests[0].gdprConsent.consentString;
-      if (request.user) {
-        if (request.user.ext) {
-          request.user.ext.consent = consentString;
+        const consentString = gdprConsent.consentString;
+        if (request.user) {
+          if (request.user.ext) {
+            request.user.ext.consent = consentString;
+          } else {
+            request.user.ext = { consent: consentString };
+          }
         } else {
-          request.user.ext = { consent: consentString };
+          request.user = { ext: { consent: consentString } };
```

Comment:
Given the series of nested object assignments we're using in this part of the file, would it make sense to create a new `utils` function similar to `deepAccess` where it will go through the object chain provided and try to assign the value to the deepest level that exists?  This would help simplify some of the code around here.